modules/preview-react/segmented-control/lib/SegmentedControlItem.tsx

```diff
@@ -52,147 +49,188 @@ export interface ItemProps extends ButtonContainerProps {
-      label: colors.blackPepper400,
-    },
-  };
-};
+    [buttonStencil.vars.borderRadius]: system.shape.x1,
+    [buttonStencil.vars.label]: system.color.static.gray.strong,
+    [systemIconStencil.vars.color]: system.color.static.gray.strong,
 
-const getPaddingStyles = (
-  children: React.ReactNode,
-  size: ButtonContainerProps['size'],
-  icon: CanvasSystemIcon | undefined
-) => {
-  if (!children) {
-    return 0;
-  }
+    '&:hover, &.hover': {
+      [buttonStencil.vars.background]: system.color.bg.alt.strong,
+      [buttonStencil.vars.label]: system.color.text.strong,
+      [systemIconStencil.vars.color]: system.color.text.strong,
+    },
 
-  switch (size) {
-    case 'large':
-      return icon ? `0 ${space.m} 0 20px` : `0 ${space.m}`;
+    '&:active, &.active': {
+      [buttonStencil.vars.background]: system.color.bg.alt.strong,
+      [buttonStencil.vars.label]: system.color.text.strong,
+      [systemIconStencil.vars.color]: system.color.text.strong,
+    },
 
-    case 'medium':
-      return icon ? `0 20px 0 ${space.s}` : `0 ${space.s}`;
+    '&:focus-visible, &.focus': {
+      [buttonStencil.vars.label]: system.color.static.gray.strong,
+      [systemIconStencil.vars.color]: system.color.static.gray.strong,
+    },
 
-    case 'small':
-      return icon ? `0 ${space.xs} 0 ${space.xxs}` : `0 ${space.xs}`;
+    '&:disabled, &.disabled': {
+      [buttonStencil.vars.background]: system.color.bg.alt.soft,
+      [buttonStencil.vars.label]: system.color.static.gray.strong,
+      [systemIconStencil.vars.color]: system.color.static.gray.strong,
+    },
 
-    default:
-      return icon ? `0 20px 0 ${space.s}` : `0 ${space.s}`;
-  }
-};
+    "&[aria-pressed='true']": {
+      [buttonStencil.vars.background]: system.color.bg.default,
+      [buttonStencil.vars.border]: system.color.border.input.default,
+      [systemIconStencil.vars.color]: system.color.text.strong,
+      [buttonStencil.vars.label]: system.color.text.strong,
 
-const geButtonStyles = (size: ButtonSizes, children: React.ReactNode, icon?: CanvasSystemIcon) => {
-  const buttonSize = getButtonSize(size);
-  const minWidthValue = getMinWidthStyles(children, children ? size : buttonSize);
+      '&:hover, &.hover': {
+        [systemIconStencil.vars.color]: system.color.text.strong,
+        [buttonStencil.vars.label]: system.color.text.strong,
+      },
 
-  return {
-    height: getMinWidthStyles(false, buttonSize),
-    minWidth: minWidthValue,
-    padding: getPaddingStyles(children, size, icon),
-  };
-};
+      '&:disabled, &.disabled': {
+        [buttonStencil.vars.border]: system.color.border.input.default,
+        [buttonStencil.vars.label]: system.color.text.strong,
+        [systemIconStencil.vars.color]: system.color.text.strong,
+      },
+    },
+    ':dir(rtl)': {
+      svg: {
+        transform: 'scaleX(-1)',
+      },
+    },
+  },
+  modifiers: {
+    size: {
+      large: {
+        height: system.space.x10,
+        paddingBlock: px2rem(9),
```

Comment:
what changes for us to have weird padding numbers?

Reply:
Actually, we don't need them at all as we set height and minWidth values